chore(github): Add template for issues and pull requests

arjunan-k opened the pull request:
## Summary

Added templates for Feature Request, Bug Report, Questions and Pull Request, as suggested by @sandrina-p in issue no #13.

## Changes Made

Added the corresponding files in the following paths:
```txt
json-schema-form/
│
├── .github/
│   ├── ISSUE_TEMPLATE/
│   │   ├── feature_request.md
│   │   ├── bug_report.md
│   │   ├── question.md
│   ├── PULL_REQUEST_TEMPLATE/
│   │   ├── pull_request_template.md
│
├── ... (other project files and folders)
```

## Screenshots

### Feature Request

![Screenshot (3)](https://github.com/remoteoss/json-schema-form/assets/104669486/59794a01-d86d-403b-a2e0-6a31ce6dc130)

### Bug Report

![Screenshot (2)](https://github.com/remoteoss/json-schema-form/assets/104669486/7fc2fb60-8b86-48b4-b863-7b79291b6476)

### Ask a Question

![Screenshot (4)](https://github.com/remoteoss/json-schema-form/assets/104669486/8c2a860d-fabe-4c6e-bd16-d646c3d40007)

### Pull Request

![Screenshot (1)](https://github.com/remoteoss/json-schema-form/assets/104669486/d9ce70e5-6996-452a-b03f-5b0b8058882d)

Hi @sandrina-p and @brennj 👋, could either of you please review this PR? Thank you 😊!

sandrina-p commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,38 @@
+
+Provide a concise and clear summary of your pull request.
+
+## Motivation
```
suggestion: I believe the "Motion" isn't needed in favor of the Related Issues/PRs which by itself contains the reason behind the PR. What do you think?

arjunan-k replied:
Feels the same now. Makes it much more minimal.

sandrina-p commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,38 @@
+
+### Mobile
+
+### iPad
```
Suggestion: No need for these Desktop / mobile / iPad sections because this tool is headless (no UI provided). :)

arjunan-k replied:
Just went with the flow of standard PR templates in frontend projects 😅
Removed 👍🏼

sandrina-p commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,38 @@
+
+## Checklist
+
+- [x] I have read the [Contributing Guidelines](../../CONTRIBUTING.md) and followed them.
+- [ ] I have tested my changes thoroughly.
+- [ ] My code follows the project's coding standards and style guide.
+- [ ] I have updated documentation (if applicable).
```
```suggestion
- [x] I have read the [Contributing Guidelines](../../CONTRIBUTING.md) and followed them.
- [ ] Unit tests were added.
```

- I removed the "coding standards" part because it's already covered by eslint and prettier on CI (automated check)
- I removed the documentation because currently, that lives outside the repo as explained in the Contributing file.

arjunan-k replied:
I have made the suggested changes.

sandrina-p commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,38 @@
+- [ ] My code follows the project's coding standards and style guide.
+- [ ] I have updated documentation (if applicable).
+
+## Any Other Context
```
suggestion: I find this section not very useful. Personally, I'd expect the authors to add this directly in the "changes made" or as part of the PR comments. What do you think?

sandrina-p reviewed (changes requested):
Hi @arjunan-k , thanks for your PR! Review done :)

sandrina-p commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,47 @@
+# Bug Report
+
```
Unrelated to code, but the PR itself. Could you rename the title from

```diff
Template: add template for issues and pull requests
chore(github): Add template for issues and pull requests
```

To follow the [conventional commits mentioned in the contributing file](https://github.com/remoteoss/json-schema-form/blob/main/CONTRIBUTING.md#merging-a-pr)

arjunan-k replied:
Added a section on commit structure to the documentation, and included a clear example for reference.

arjunan-k commented:
> Hi @arjunan-k , thanks for your PR! Review done :)

Hi @sandrina-p, Thank you for your review and feedback. I've addressed the comments in the PR. Would you kindly review the updated PR at your convenience : )
